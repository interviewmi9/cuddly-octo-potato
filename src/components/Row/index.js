import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

export const justifyOptions = [
  'flex-start',
  'center',
  'flex-end',
  'space-between',
  'space-around',
]
export const alignOptions = [
  'flex-start',
  'center',
  'flex-end',
  'stretch',
  'baseline',
]

const Layout = styled.div`
  justify-content: ${props => props.justify};
  align-items: ${props => props.align};
  display: flex;
  flex-direction: row;
  width: 100%;
  flex: 1;
  & > * {
    margin: 0 ${props => props.spacing}px;
    &:first-child {
      margin-left: 0;
    }
    &:last-child {
      margin-right: 0;
    }
  }
`
const Row = ({ children, ...props }) => <Layout {...props}>{children}</Layout>

Row.propTypes = {
  align: PropTypes.oneOf(alignOptions),
  children: PropTypes.node.isRequired,
  justify: PropTypes.oneOf(justifyOptions),
  spacing: PropTypes.number,
}
Row.defaultProps = {
  align: 'flex-start',
  justify: 'flex-start',
  spacing: 10,
}

export default Row
