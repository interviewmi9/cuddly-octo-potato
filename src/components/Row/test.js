import React from 'react'
import Row from './'

describe('Row component', () => {
  it('renders correctly', () => {
    expect(render(<Row>Hello</Row>)).toMatchSnapshot()
  })
})
