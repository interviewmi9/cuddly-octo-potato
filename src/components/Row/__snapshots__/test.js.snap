// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component renders correctly 1`] = `
.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 > * {
  margin: 0 10px;
}

.c0 > *:first-child {
  margin-left: 0;
}

.c0 > *:last-child {
  margin-right: 0;
}

<div
  className="c0"
  spacing={10}
>
  Hello
</div>
`;
